src/components/Autocomplete/components/ComboBox/tests/ComboBox.test.tsx

```diff
@@ -418,82 +340,204 @@ describe('<ComboBox/>', () => {
   });
 
   describe('keypress events', () => {
-    it('selects the first option when the down arrow is pressed', () => {
-      const comboBox = mountWithAppProvider(
-        <ComboBox
-          options={options}
-          selected={[]}
-          textField={renderTextField()}
-          onSelect={noop}
-        />,
-      );
-
-      async () => {
-        comboBox.find(TextField).simulate('click');
-        listenerMap.keyup({keyCode: Key.DownArrow});
-        await expect(comboBox.state('selectedIndex')).toBe(0);
-      };
-    });
-
-    it('adds to selected options when the down arrow and enter keys are pressed', () => {
-      const spy = jest.fn();
-      const comboBox = mountWithAppProvider(
-        <ComboBox
-          options={options}
-          selected={[]}
-          textField={renderTextField()}
-          onSelect={spy}
-        />,
-      );
-
-      async () => {
-        comboBox.find(TextField).simulate('click');
-        await listenerMap.keyup({keyCode: Key.DownArrow});
-      };
-
-      async () => {
-        listenerMap.keyup({keyCode: Key.Enter});
-        await expect(spy).toHaveBeenCalledTimes(1);
-        expect(comboBox.prop('selected')[0]).toBe('cheese_pizza');
-      };
-    });
-
-    it('activates the popover when the tab key is pressed', () => {
-      const comboBox = mountWithAppProvider(
-        <ComboBox
-          options={options}
-          selected={[]}
-          textField={renderTextField()}
-          onSelect={noop}
-        />,
-      );
-
-      async () => {
-        listenerMap.keyup({keyCode: Key.Tab});
-        await expect(comboBox.state('popoverActive')).toBe(true);
-      };
-    });
-
-    it('deactivates the popover when the escape key is pressed', () => {
-      const comboBox = mountWithAppProvider(
-        <ComboBox
-          options={options}
-          selected={[]}
-          textField={renderTextField()}
-          onSelect={noop}
-        />,
-      );
-
-      async () => {
-        comboBox.find(TextField).simulate('click');
-        await listenerMap.keyup({keyCode: Key.DownArrow});
-        expect(comboBox.state('popoverActive')).toBe(true);
-      };
-
-      async () => {
-        listenerMap.keyup({keyCode: Key.Escape});
-        await expect(comboBox.state('popoverActive')).toBe(false);
-      };
+    describe('down arrow', () => {
+      it('selects nothing if there are no options', () => {
+        const options: OptionDescriptor[] = [];
+        const comboBox = mountWithAppProvider(
+          <ComboBox {...mockProps} options={options} />,
+        );
+
+        async () => {
```

Comment:
If this is `async` you probably wanna just make the whole test an `async` test.

```ts
it('selects nothing if there are no options', async () => {});
```

Reply:
Typescript doesn't like comboBox being declared within the async function.

Reply:
What happens when you do?

Reply:
Why do we need `async/await` in these tests? I don't think the code in the anonymous function is being executed, you probably want a `IIFE`. But as  Koen suggested adding `async` to the `it` block is preferred.